convert_jendl.py

```diff
@@ -0,0 +1,139 @@
+# the destination is decided after the release is known to avoid putting the release in a folder with a misleading name
+if args.destination == None:
+    args.destination = '-'.join([library_name, args.release, 'hdf5'])
+
+# This dictionary contains all the unique information about each release. This can be exstened to accommodated new releases
+release_details = {
+    '4.0': {
+        'base_url': 'https://wwwndc.jaea.go.jp/ftpnd/ftp/JENDL/',
+        'files': ['jendl40-or-up_20160106.tar.gz'],
+        'neutron_files': os.path.join(endf_files_dir, 'jendl40-or-up_20160106', '*.dat'),
+        'metastables': os.path.join(endf_files_dir, 'jendl40-or-up_20160106', '*m.dat'),
+        'compressed_file_size': '0.2 GB',
+        'uncompressed_file_size': '? GB'
+    }
+}
+
+download_warning = """
+WARNING: This script will download {} of data.
+Extracting and processing the data requires {} of additional free disk space.
+
+Are you sure you want to continue? ([y]/n)
+""".format(release_details[args.release]['compressed_file_size'],
+           release_details[args.release]['uncompressed_file_size'])
+
+response = input(download_warning) if not args.batch else 'y'
+if response.lower().startswith('n'):
+    sys.exit()
+
+block_size = 16384
+
+# ==============================================================================
+# DOWNLOAD FILES FROM WEBSITE
+
+files_complete = []
+for f in release_details[args.release]['files']:
+    # Establish connection to URL
+    url = release_details[args.release]['base_url'] + f
+    downloaded_file = download(url)
+    files_complete.append(downloaded_file)
+
+# ==============================================================================
+# EXTRACT FILES FROM TGZ
+
+for f in release_details[args.release]['files']:
+    if f not in files_complete:
+        continue
+
+    # Extract files
+
+    suffix = ''
+    with tarfile.open(f, 'r') as tgz:
+        print('Extracting {0}...'.format(f))
+        tgz.extractall(path=os.path.join(endf_files_dir, suffix))
+
+
+# ==============================================================================
+# GENERATE HDF5 LIBRARY -- NEUTRON FILES
+
+# Get a list of all ACE files
+neutron_files = glob.glob(release_details[args.release]['neutron_files'])
+
+# Create output directory if it doesn't exist
+if not os.path.isdir(args.destination):
+    os.mkdir(args.destination)
+
+library = openmc.data.DataLibrary()
+
+for filename in sorted(neutron_files):
+
+    # this is a fix for the TENDL-2017 release where the B10 ACE file which has an error on one of the values
+    if library_name == 'tendl' and args.release == '2017' and os.path.basename(filename) == 'B010':
+        text = open(filename, 'r').read()
+        if text[423:428] == '86843':
+            print('Manual fix for incorrect value in ACE file') # see OpenMC user group issue for more details
+            text = ''.join(text[:423])+'86896'+''.join(text[428:])
+            open(filename, 'w').write(text)
```

Comment:
This block here modifying B10 seems unnecessary for JENDL.

Reply:
Oh this is really bad, sorry for that